### PR TITLE
feat(math): create point from x coordinate if (x, y) belongs to a curve point

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR implements point creation from x coordinate using #90.
This also adds missing `IsOnCurve` tests to `JacobianPoint`, `ProjectivePoint` and `PointXYZZ`.

### CODE REVIEW

File: tachyon/math/elliptic_curves/short_weierstrass/affine_point.h [hunk 1]
```diff
@@ -61,6 +62,13 @@ class AffinePoint<_Curve, std::enable_if_t<_Curve::kIsSWCurve>> final
     return ret;
   }
 
+  constexpr static std::optional<AffinePoint> CreateFromX(const BaseField& x,
+                                                          bool pick_bigger) {
+    AffinePoint point;
+    if (!Curve::GetPointFromX(x, pick_bigger, &point)) return std::nullopt;
```
Comment (dongchangYoo) [inline tachyon/math/elliptic_curves/short_weierstrass/affine_point.h:68]: Typically, when selecting a `Point`, one cannot determine whether their y-value is large or small (requiring additional computation), so in Ethereum signatures, the y-value is chosen based on "odd/even." Don't you think we need to address this?

File: tachyon/math/elliptic_curves/short_weierstrass/sw_curve.h [hunk 2]
```diff
@@ -37,6 +40,45 @@ class SWCurve {
     Config::Init();
   }
 
+  // Attempts to construct an affine point given an |x| coordinate. The
+  // point is not guaranteed to be in the prime order subgroup.
+  // If |pick_bigger| is set to true, it chooses a bigger y coordinate.
+  template <typename PointTy>
+  constexpr static bool GetPointFromX(const BaseField& x, bool pick_bigger,
```
Comment (dongchangYoo) [inline tachyon/math/elliptic_curves/short_weierstrass/sw_curve.h:47]: Typically, when selecting a Point, one cannot determine whether their y-value is large or small (requiring additional computation), so in Ethereum signatures, the y-value is chosen based on "odd/even." Don't you think we need to address this?

Comment (chokobole) [reply tachyon/math/elliptic_curves/short_weierstrass/sw_curve.h:47]: I took it from arkworks, but I think your suggestion is nicer!

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**